crates/bevy_pbr/src/prepass/prepass.wgsl

```diff
@@ -75,12 +75,31 @@ fn morph_vertex(vertex_in: Vertex) -> Vertex {
     }
     return vertex;
 }
+
+
+#ifdef MOTION_VECTOR_PREPASS
+fn morph_previous_position(vertex: Vertex) -> vec3<f32> {
+    var vertex_position = vertex.position;
+    let weight_count = bevy_pbr::morph::layer_count();
+    for (var i: u32 = 0u; i < weight_count; i++) {
+        let weight = bevy_pbr::morph::previous_weight_at(i);
+        if weight != 0.0 {
+            vertex_position += weight * bevy_pbr::morph::morph(vertex.index, bevy_pbr::morph::position_offset, i);
+        }
+    }
+    return vertex_position;
+}
+#endif // MOTION_VECTOR_PREPASS
 #endif
 
 @vertex
 fn vertex(vertex_no_morph: Vertex) -> VertexOutput {
     var out: VertexOutput;
 
+    // Use vertex_no_morph.instance_index instead of vertex.instance_index to work around a wgpu dx12 bug.
+    // See https://github.com/gfx-rs/naga/issues/2416
+    var vertex_instance = vertex_no_morph.instance_index;
```

Comment:
https://github.com/gfx-rs/naga/issues/2416 seems to say it should work if the binding is a `var`? an alternative is to declare the instance index as an argument to `fn vertex` instead of field of `vertex` struct.

Reply:
Iirc we specifically didn't bind `vertex_no_morph.instance_index` to a variable despite it being (slightly) cleaner because the codegen was worse.

Reply:
I updated the code to pass `vertex_instance` as a second parameter to `vertex` instead of part of the `Vertex` struct, is that sufficient?

Reply:
Tested it and it seems to work fine.